content/tutorials/use-specialized-payment-types/create-custom-transactors.md

```diff
@@ -0,0 +1,379 @@
+
+    sleCheck->setAccountID(sfAccount, account_);
+    AccountID const dstAccountId = ctx_.tx[sfDestination];
+    sleCheck->setAccountID(sfDestination, dstAccountId);
+    sleCheck->setFieldU32(sfSequence, seq);
+    sleCheck->setFieldAmount(sfSendMax, ctx_.tx[sfSendMax]);
+    if (auto const srcTag = ctx_.tx[~sfSourceTag])
+        sleCheck->setFieldU32(sfSourceTag, *srcTag);
+    if (auto const dstTag = ctx_.tx[~sfDestinationTag])
+        sleCheck->setFieldU32(sfDestinationTag, *dstTag);
+    if (auto const invoiceId = ctx_.tx[~sfInvoiceID])
+        sleCheck->setFieldH256(sfInvoiceID, *invoiceId);
+    if (auto const expiry = ctx_.tx[~sfExpiration])
+        sleCheck->setFieldU32(sfExpiration, *expiry);
+
+    view().insert(sleCheck);
+
+    auto viewJ = ctx_.app.journal("View");
+    // If it's not a self-send (and it shouldn't be), add Check to the
+    // destination's owner directory.
+    if (dstAccountId != account_)
+    {
+        auto const page = view().dirInsert(
+            keylet::ownerDir(dstAccountId),
+            checkKeylet,
+            describeOwnerDir(dstAccountId));
+
+        JLOG(j_.trace()) << "Adding Check to destination directory "
+                        << to_string(checkKeylet.key) << ": "
+                        << (page ? "success" : "failure");
+
+        if (!page)
+            return tecDIR_FULL;
+
+        sleCheck->setFieldU64(sfDestinationNode, *page);
+    }
+
+    {
+        auto const page = view().dirInsert(
+            keylet::ownerDir(account_),
+            checkKeylet,
+            describeOwnerDir(account_));
+
+        JLOG(j_.trace()) << "Adding Check to owner directory "
+                        << to_string(checkKeylet.key) << ": "
+                        << (page ? "success" : "failure");
+
+        if (!page)
+            return tecDIR_FULL;
+
+        sleCheck->setFieldU64(sfOwnerNode, *page);
+    }
+    // If we succeeded, the new entry counts against the creator's reserve.
+    adjustOwnerCount(view(), sle, 1, viewJ);
+    return tesSUCCESS;
+}
+```
+
+
+## Additional Functions
+
+
+### `calculateBaseFee`
+
+The `calculateBaseFee` function calculates the XRP fee to run a transaction. The fee is usually inherited from the base transactor, but transactors such as `EscrowFinish` require additional fees.
+
+```c++
+XRPAmount
+EscrowFinish::calculateBaseFee(ReadView const& view, STTx const& tx)
+{
+    XRPAmount extraFee{0};
+
+    if (auto const fb = tx[~sfFulfillment])
+    {
+        extraFee += view.fees().base * (32 + (fb->size() / 16));
+    }
+
+    return Transactor::calculateBaseFee(view, tx) + extraFee;
+}
+```
+
+
+### `makeTxConsequences`
+
+`rippled` uses a [`TxConsequences`](https://github.com/XRPLF/rippled/blob/master/src/ripple/app/tx/applySteps.h#L41-L44) class to describe the outcome to an account when applying a transaction. It tracks the fee, maximum possible XRP spent, and how many sequences are consumed by the transaction. There are three types of consequences:
```

Comment:
Based on what I'm seeing here, I have a feeling the point of this function is to determine when a transaction is [eligible for the queue](https://xrpl.org/transaction-queue.html#queuing-restrictions) and how it affects the queue eligibility of transactions that come after it. I think, relevantly, this needs to be something that can be evaluated using the fields of the transaction itself without further context like the ledger state. So the function should estimate the transaction's max XRP spend, any implications for account sequence numbers and transaction authorization.

FWIW, "sequences" is not a phrasing we use elsewhere in the docs; for consistency, it's probably better to write "sequence numbers" instead.